lib/utils/local_settings.dart

```diff
@@ -13,6 +13,7 @@ class LocalSettings {
   static final LocalSettings instance = LocalSettings._privateConstructor();
   static const kCollectionSortPref = "collection_sort_pref";
   static const kPhotoGridSize = "photo_grid_size";
+  static const kHideSharedItems = "hide_shared_items";
```

Comment:
Internally, I think we should still call it `archive_shared_items`

Note:
Archived items are searchable. User can view them without any on-device auth.
Hidden items are supposed to be not searchable. User should perform on-device auth before they can view them.